 Gallery's progress bar stay at 0 when play some clips

If audio is eos before video, video should call updateMaxTimeMedia,
or mediatime will always stay at the position when audio ends.

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index 8f68993..3dbe3c0 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -112,6 +112,7 @@ NuPlayer::Renderer::Renderer(
       mVideoLateByUs(0ll),
       mHasAudio(false),
       mHasVideo(false),
+      mAudioEOS(false),
       mNotifyCompleteAudio(false),
       mNotifyCompleteVideo(false),
       mSyncQueues(false),
@@ -1212,7 +1213,7 @@ void NuPlayer::Renderer::postDrainVideoQueue() {
                 realTimeUs = nowUs;
             }
         }
-        if (!mHasAudio) {
+        if (!mHasAudio || mAudioEOS) {
             // smooth out videos >= 10fps
             mMediaClock->updateMaxTimeMedia(mediaTimeUs + 100000);
         }
@@ -1386,6 +1387,7 @@ void NuPlayer::Renderer::onQueueBuffer(const sp<AMessage> &msg) {
 
     if (audio) {
         mHasAudio = true;
+        mAudioEOS = false;
     } else {
         mHasVideo = true;
     }
@@ -1496,6 +1498,7 @@ void NuPlayer::Renderer::onQueueEOS(const sp<AMessage> &msg) {
         }
         mAudioQueue.push_back(entry);
         postDrainAudioQueue_l();
+        mAudioEOS = true;
     } else {
         if (mVideoQueue.empty() && getSyncQueues()) {
             Mutex::Autolock autoLock(mLock);
diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h
index 8f68993..3dbe3c1 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h
@@ -167,6 +167,7 @@ struct NuPlayer::Renderer : public AHandler {
     int64_t mVideoLateByUs;
     bool mHasAudio;
     bool mHasVideo;
+    bool mAudioEOS;
 
     bool mNotifyCompleteAudio;
     bool mNotifyCompleteVideo;
